include/max_element.hpp

```diff
@@ -21,11 +21,53 @@ namespace bit {
 
 
 
-// Status: to do
+// Status: complete
 template <class ForwardIt>
 constexpr bit_iterator<ForwardIt> max_element(bit_iterator<ForwardIt> first,
     bit_iterator<ForwardIt> last) {
-    (last);
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+
+    bit_iterator<ForwardIt> cursor = first;
+
+    // handle misaligned first word. for example ^00100^101
+    if (!_is_aligned_lsb(cursor)) {
+        word_type first_word = _padded_read(cursor, last, bit::bit0);
+        if (first_word > 0) {
+            std::size_t first_set_bit_position = _tzcnt(first_word);
+            return cursor + (first_set_bit_position - cursor.position());
+        }
+
+        if (_in_same_word(cursor, last) ||
+            _is_one_past_last_bit(last, cursor.base())) {
+            return first;
+        }
+
+        cursor = bit_iterator(std::next(cursor.base()));
+    }
+
+    // full words
+    while (!_in_same_word(cursor, last) ||
+        _is_one_past_last_bit(last, cursor.base())) {
+        word_type cur_word = *(cursor.base());
+        if (cur_word > 0) {
+            std::size_t cur_set_bit_position = _tzcnt(cur_word);
+            return cursor + (cur_set_bit_position - cursor.position());
+        }
+        cursor = bit_iterator(std::next(cursor.base()));
+    }
```

Comment:
Do you know if calling the constructor on every iteration is also optimized? 

If we have cursor as a word iterator, we still need to increment it at every step, so I'm not worried about that. But is there any overhead to calling the `bit_iterator` constructor on every iteration?

I think the rest of my comments on this PR are very nitpicky and don't offer much performance gain, but I am interested in this one.